### PR TITLE
Phoenix-6153 Table Map Reduce job after a Snapshot based job fails with CorruptedSnapshotException

### PR DESCRIPTION
**Pull request by sakshamgangwar**

### CODE REVIEW

**Comment by sakshamgangwar** [comment]
@ChinmaySKulkarni @dbwong I will need your review on this. Thanks.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java [hunk 1]
```diff
@@ -63,13 +71,24 @@
...
+  private static final String RECORDINGS_QUARTER = "RECORDINGS_QUARTER";
+  private static final String MAX_RECORDING = "MAX_RECORDING";
   private final static String SNAPSHOT_NAME = "FOO";
   private static final String FIELD1 = "FIELD1";
   private static final String FIELD2 = "FIELD2";
   private static final String FIELD3 = "FIELD3";
   private String CREATE_TABLE = "CREATE TABLE IF NOT EXISTS %s ( " +
       " FIELD1 VARCHAR NOT NULL , FIELD2 VARCHAR , FIELD3 INTEGER CONSTRAINT pk PRIMARY KEY (FIELD1 ))";
   private String UPSERT = "UPSERT into %s values (?, ?, ?)";
+  private static final String CREATE_STOCK_TABLE = "CREATE TABLE IF NOT EXISTS %s ( " +
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java:85]: nit: Use the variables for column names in the DDL statements as well

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java [hunk 2]
```diff
@@ -63,13 +71,24 @@
...
+          " STOCK_NAME VARCHAR NOT NULL , RECORDING_YEAR  INTEGER NOT  NULL,  RECORDINGS_QUARTER " +
+          " DOUBLE array[] CONSTRAINT pk PRIMARY KEY ( STOCK_NAME, RECORDING_YEAR )) "
+          + "SPLIT ON ('AA')";
+  private static final String CREATE_STOCK_STATS_TABLE =
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java:89]: same here

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java [hunk 3]
```diff
@@ -127,6 +146,77 @@ public void testMapReduceSnapshotWithLimit() throws Exception {
...
+  @Test
+  public void testSnapshotMapReduceJobNotImpactingTableMapReduceJob() throws Exception {
+    //Submitting and asserting successful Map Reduce Job over snapshots
+    PhoenixMapReduceUtil
+            .setInput(job, PhoenixIndexDBWritable.class, SNAPSHOT_NAME, tableName, tmpDir, null,
+                    FIELD1, FIELD2, FIELD3);
+    configureJob(job, tableName, null, null, false);
+
+    // create table
+    Connection conn = DriverManager.getConnection(getUrl());
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java:158]: Use try-with-resources when opening connections and statements throughout.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java [hunk 4]
```diff
@@ -127,6 +146,77 @@ public void testMapReduceSnapshotWithLimit() throws Exception {
...
+    // any wrong snapshot properties set in common configurations which are
+    // used across all jobs.
+    createAndTestJob(conn);
+  }
+
+  private void createAndTestJob(Connection conn)
+          throws SQLException, IOException, InterruptedException, ClassNotFoundException {
+    String stockTableName = generateUniqueName();
+    String stockStatsTableName = generateUniqueName();
+    conn.createStatement().execute(String.format(CREATE_STOCK_TABLE, stockTableName));
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java:173]: ditto about try-with-resources

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java [hunk 5]
```diff
@@ -170,6 +260,22 @@ private void configureJob(Job job, String tableName, String inputQuery, String c
     }
   }
 
+  private void upsertData(Connection conn, String stockTableName) throws SQLException {
+    PreparedStatement stmt = conn.prepareStatement(String.format(UPSERT, stockTableName));
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java:264]: ditto about try-with-resources

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java [hunk 6]
```diff
@@ -127,6 +146,77 @@ public void testMapReduceSnapshotWithLimit() throws Exception {
...
+    job.setReducerClass(MapReduceIT.StockReducer.class);
+    job.setOutputFormatClass(PhoenixOutputFormat.class);
+
+    job.setMapOutputKeyClass(Text.class);
+    job.setMapOutputValueClass(DoubleWritable.class);
+    job.setOutputKeyClass(NullWritable.class);
+    job.setOutputValueClass(MapReduceIT.StockWritable.class);
+
+    // run job and assert if success
+    assertTrue("Job didn't complete successfully! Check logs for reason.", job.waitForCompletion(true));
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java:204]: Shouldn't we also get the final job conf here and test that it doesn't contain any snapshot related properties? Is that possible?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java (hunk 2 above)
Comment (sakshamgangwar) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java:90]: @ChinmaySKulkarni I have updated it.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java (hunk 4 above)
Comment (sakshamgangwar) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java:184]: Handled in try-with-resources block in calling method.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java (hunk 6 above)
Comment (sakshamgangwar) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java:216]: @ChinmaySKulkarni Added the assertion both after snapshot MR job and after Table based job too.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java (hunk 5 above)
Comment (sakshamgangwar) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java:276]: handled in the caller test method.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java (hunk 1 above)
Comment (sakshamgangwar) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java]: done

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java (hunk 3 above)
Comment (sakshamgangwar) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java]: using it now

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java (hunk 4 above)
Comment (ChinmaySKulkarni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java:184]: Please also close statements everywhere

**Comment by yanxinyi** [comment]
lgtm +1

**Comment by gjacoby126** [comment]
JIRA is marked resolved, closing PR

**Pull request closed without merging**